### DISCUSSION TITLE
Save Button for Form and search Data?

### DISCUSSION THREAD

**Discussion by Gehmalbierholen123e** [Budibase/budibase#10071] [Help] [answered]
Hello,
I'm sorry if this is a beginner question, but I can't find an answer to it.
Here is what I want to do:
Having a form to add information about persons which will be added to the database. Then be able to search the Data with a Search Bar, filter options and more.
I have now created one screen with a Form that has all the necessary fields and another screen where I want the search to be located.
2 Problems:
1. The Form has no send button to send the Data. How do I add one?#
2. How can I add a search section on my screen, where I can search all entries and display them in an organized way? I looked on the help page in the docs but couldn't really figure it out.

Thank you for anyone who can help me  :)

**Comment by ConorWebb96** [comment, marked as answer]
Hey @Gehmalbierholen123e,

Firstly welcome! 

1. A **Form block** may be exactly what you want to use in this case, all you will need to do is add this component then select the table you wish to use it will generate all the fields needed for you as well as the button action. You can then eject this block and see all fields and data providers used. (Screenshots below)

https://docs.budibase.com/docs/form-block 

<img width="937" alt="Screenshot 2023-03-20 at 12 47 17" src="https://user-images.githubusercontent.com/126772285/226344170-2b44afbc-315b-4e8a-b8e6-b41416814461.png">
<img width="366" alt="Screenshot 2023-03-20 at 12 47 30" src="https://user-images.githubusercontent.com/126772285/226344177-0f63338e-7a94-413d-953d-a94227ab25d2.png">
<img width="251" alt="Screenshot 2023-03-20 at 12 51 39" src="https://user-images.githubusercontent.com/126772285/226344180-a1825d38-2474-4695-a10a-1e8aea6cac7b.png">

2. This can be a little complicated depending on what you want to do. If you are using the **table block** component this is quite simple as all you need to do is enable searching and you are ready to go. (I've attached screenshots below on everything discussed here.)

https://docs.budibase.com/docs/table-block

<img width="653" alt="Screenshot 2023-03-20 at 12 44 54" src="https://user-images.githubusercontent.com/126772285/226342650-6a6f89a8-858e-44dd-bdad-926c05ae59ca.png">
<img width="254" alt="Screenshot 2023-03-20 at 12 45 14" src="https://user-images.githubusercontent.com/126772285/226342656-5f7da96e-71a9-4a47-9e38-20b4bcc37064.png">
<img width="1383" alt="Screenshot 2023-03-20 at 12 45 40" src="https://user-images.githubusercontent.com/126772285/226342664-6849cb58-abe8-428c-974a-1c8232c1719b.png">

I hope this helped point you in the right direction!

**Reply by Gehmalbierholen123e** [reply]
@ConorWebb96 Hi,
I have now figured everything out and it's working as expected.  Thank you very much!
